### PR TITLE
[doc] Add the workaround for duplicated samples to FAQ

### PR DESCRIPTION
**Pull request by nabenabe0928**
## Motivation

This PR addresses the issue raised in #4741.

## Description of the changes

The issue mentioned that Optuna sometimes duplicates parameters evaluated in the past and I added the workaround to the FAQ.

### CODE REVIEW

**Comment by not522** [comment]
@contramundum53 Could you review this PR?

File: docs/source/faq.rst [hunk 1]
```diff
@@ -682,3 +682,30 @@ An Optuna implementation example to solve Euclid TSP is as follows:
...
+    from optuna.trial import TrialState
+
+
+    def objective(trial):
+        # Sample parameters
+        x = trial.suggest_int("x", -5, 5)
+        y = trial.suggest_int("x", -5, 5)
+        # Fetch all the all the completed trials
+        complete_trials = trial.study.get_trials(deepcopy=False, states=(TrialState.COMPLETE,))
+        # Check whether we already evaluated the sampled `x`
```
Comment (nzw0301) [inline docs/source/faq.rst:703]: I suppose this line also checks `y`:
```suggestion
        # Check whether we already evaluated the sampled `x` and `y`.
```

File: docs/source/faq.rst [hunk 2]
```diff
@@ -682,3 +682,30 @@ An Optuna implementation example to solve Euclid TSP is as follows:
...
+
+    import optuna
+    from optuna.trial import TrialState
+
+
+    def objective(trial):
+        # Sample parameters
+        x = trial.suggest_int("x", -5, 5)
+        y = trial.suggest_int("x", -5, 5)
+        # Fetch all the all the completed trials
```
Comment (nzw0301) [inline docs/source/faq.rst:701]: Typo?
```suggestion
        # Fetch all the completed trials
```

File: docs/source/faq.rst [hunk 3]
```diff
@@ -682,3 +682,30 @@ An Optuna implementation example to solve Euclid TSP is as follows:
...
+Optuna may sometimes suggest parameters evaluated in the past and if you would like to avoid this problem, you can try out the following workaround:
+
+.. code-block:: python
+
+    import optuna
+    from optuna.trial import TrialState
+
+
+    def objective(trial):
+        # Sample parameters
```
Comment (nzw0301) [inline docs/source/faq.rst:698]: By following Optna's coding convention, we need to add `.` at the end of sentence:
```suggestion
        # Sample parameters.
```
Ref https://github.com/optuna/optuna/wiki/Coding-Style-Conventions#block-and-inline-comments-are-supposed-to-be-sentences

**Review by nzw0301 — COMMENTED** [review]
Thank you for updating the PR. Regarding my suggestion, LGTM. 

I'm curious about the other trial states beyond `COMPLETE` with duplicated parameters. Will such parameters be suggested? If so, should we treat the duplicated trial similarly? I feel that it would be nice to mention it briefly at least.

**Comment by contramundum53** [comment]
Originally I wanted to keep the code simple by ignoring RUNNING trials. Maybe it's worth adding some explanation.

**Comment by contramundum53** [comment]
(Another option is to fail trials that are duplicated with RUNNING trials.)

**Comment by nzw0301** [comment]
@contramundum53 Thank you for your comment! I agree with you; the treatment of duplicated param is user intention's dependent, so adding some explanations will be enough for this FAQ section.

File: docs/source/faq.rst [hunk 4]
```diff
@@ -682,3 +682,33 @@ An Optuna implementation example to solve Euclid TSP is as follows:
...
+        # Sample parameters.
+        x = trial.suggest_int("x", -5, 5)
+        y = trial.suggest_int("y", -5, 5)
+        # Fetch all the trials to consider.
+        # In this example, we use only completed trials, but users can specify other states
+        # such as TrialState.PRUNED and TrialState.FAILED.
+        states_to_consider = (TrialState.COMPLETE,)
+        trials_to_consider = trial.study.get_trials(deepcopy=False, states=states_to_consider)
+        # Check whether we already evaluated the sampled `(x, y)`.
+        for t in reverse(complete_trials):
```
Comment (not522) [inline docs/source/faq.rst:707]:
```suggestion
        for t in reversed(trials_to_consider):
```

File: docs/source/faq.rst [hunk 5]
```diff
@@ -682,3 +682,37 @@ An Optuna implementation example to solve Euclid TSP is as follows:
...
+    from optuna.trial import TrialState
+
+
+    def objective(trial):
+        # Sample parameters.
+        x = trial.suggest_int("x", -5, 5)
+        y = trial.suggest_int("y", -5, 5)
+        # Fetch all the trials to consider.
+        # In this example, we use only completed trials, but users can specify other states
+        # such as TrialState.PRUNED and TrialState.FAILED.
```
Comment (not522) [inline docs/source/faq.rst:703]: LGTM except for this fix.

```suggestion
        # such as TrialState.PRUNED and TrialState.FAIL.
```

**Review by not522 — APPROVED** [review]
Thank you for your update. LGTM!

**Review by contramundum53 — APPROVED** [review]
LGTM.

**Pull request merged**